src/Control/Session.php

```diff
@@ -319,36 +320,38 @@ public function start(HTTPRequest $request)
                 session_save_path($session_path);
             }
 
-            // If we want a secure cookie for HTTPS, use a seperate session name. This lets us have a
-            // seperate (less secure) session for non-HTTPS requests. Note that if this causes problems
+            // If we want a secure cookie for HTTPS, use a separate session name. This lets us have a
+            // separate (less secure) session for non-HTTPS requests
             // if headers_sent() is true then it's best to throw the resulting error rather than risk
             // a security hole.
             if ($secure) {
                 session_name($this->config()->get('cookie_name_secure'));
             }
 
-            session_start();
+            $sessionParameters = [
+                "cookie_path" => $path,
+                "cookie_domain" => $domain ?: "",
+                "cookie_secure" => $secure,
+                "cookie_httponly" => true
+            ];
+
+            if ($timeout) {
+                $sessionParameters["cookie_lifetime"] = $timeout;
+            }
+
+            session_start($sessionParameters);
```

Comment:
I'm not sure if we need to overwrite the cookie pararmeters in here at all, since we already pass them all into `session_set_cookie_params` above in the line 306?

Reply:
Yeah, it's true, it's redundant, it's the PHP7 way of setting session parameters, so we can either use the new and old or just retain the old -> `session_set_cookie_params` I didn't initially call this, but it was suggested for backwards compatibility. (5.6)